doc/HLD_of_Motr_Object_Index.rst

```diff
@@ -0,0 +1,338 @@
+=========================================
+High level design of a Motr Object Index
+=========================================
+
+This document presents a high level design (HLD) of an Object Index for Motr M0 core. The main purposes of this document are: (i) to be inspected by M0 architects and peer designers to ascertain that high level design is aligned with M0 architecture and other designs, and contains no defects, (ii) to be a source of material for Active Reviews of Intermediate Design (ARID) and detailed level design (DLD) of the same component, (iii) to serve as a design reference document.
+
+The intended audience of this document consists of M0 customers, architects, designers and developers.
+
+**************
+Introduction
+**************
+
+The object index performs the function of a metadata layer on top of M0 storage objects. A M0 storage object (stob) is a flat address space where one can read from or write on, with block size granularity. Stobs have no metadata associated with them. To be used as files (or components of files, aka stripes), additional metadata must be associated with the stobs.
```

Comment:
s
/where one can read from or write on
where one can read from or write to

Reply:
Comments Incorporated.